### PR TITLE
Add `Dice` segmentation loss

### LINKED ISSUE #586: Experiment with more sophisticated loss functions

### PR DESCRIPTION
**Pull request by ioangatop**
Closes #586 

CoNSePT: orange (bad): cross entropy, orange (good): dice online, green: dice offline
Screenshot 2024-07-25 at 3 44 18 PM

### CODE REVIEW

**Review by roman807 — APPROVED** [review]
thanks!
